src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/ThreadPoolConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+public class ThreadPoolConfig {
+    private static final int DEFAULT_THREAD_POOL_SIZE = 10;
+    private static final int DEFAULT_THREAD_POOL_MAX_QUEUE_LEN = 100_000;
+
+    private final int threadPoolSize;
+    private final int threadPoolMaxQueueLen;
+
+    public ThreadPoolConfig(int threadPoolSize, int threadPoolMaxQueueLen) {
+        if (threadPoolSize <= 0) {
+            throw new IllegalArgumentException("Incorrect thread pool size: " + threadPoolSize + " configured for audit logging.");
+        }
+
+        if (threadPoolMaxQueueLen <= 0) {
+            throw new IllegalArgumentException("Incorrect thread pool queue length: " + threadPoolMaxQueueLen + " configured for audit logging.");
+        }
+
+        this.threadPoolSize = threadPoolSize;
+        this.threadPoolMaxQueueLen = threadPoolMaxQueueLen;
+    }
+
+    public int getThreadPoolSize() {
+        return threadPoolSize;
+    }
+
+    public int getThreadPoolMaxQueueLen() {
+        return threadPoolMaxQueueLen;
+    }
+
+    public static ThreadPoolConfig getConfig(Settings settings) {
+        int threadPoolSize = settings.getAsInt(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_THREADPOOL_SIZE, DEFAULT_THREAD_POOL_SIZE);
+        int threadPoolMaxQueueLen = settings.getAsInt(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_THREADPOOL_MAX_QUEUE_LEN, DEFAULT_THREAD_POOL_MAX_QUEUE_LEN);
+
+        if (threadPoolSize <= 0) threadPoolSize = DEFAULT_THREAD_POOL_SIZE;
```

Comment:
The check is not necessary. If there are no settings, it will return a default value, otherwise, it should check validity.

Reply:
A lot of tests are setting this value to be 0 with the expectation to use the default thread pool

Reply:
There are only 3 tests that incorrectly set the value to be 0. If they want to rely on the default value, they should either set it explicitly to the default value or not to set it at all.

Reply:
I m really torn on changing this because I agree with you but removing this will change the functionality 
- previous: if set <= 0 or not defined, use default 
- current: if set <= 0, complain with exception. if not defined, use default.

Changing functionality is something normally I wouldn't do in a refactor. 
There are actually 40+ tests explicitly setting: opendistro_security.audit.threadpool.size to 0. 
If you think changing the functionality is the correct way, let me know. I will change it.

Reply:
I suggest changing the behavior to avoid the possibility of overruling explicitly set by a user settings. IMO, tests need to be changed anyway to avoid using `"opendistro_security.audit.threadpool.size"` string instead of `OPENDISTRO_SECURITY_AUDIT_THREADPOOL_SIZE` constant. If a test needs to use **not a default value** it should use `ConfigConstants.OPENDISTRO_SECURITY_AUDIT_THREADPOOL_SIZE`.